lib/tasks/bower-install.js

```diff
@@ -1,32 +1,53 @@
 'use strict';
 
 // Runs `bower install` in cwd
+const fs = require('fs-extra');
 const path = require('path');
 const existsSync = require('exists-sync');
 const execa = require('execa');
 const RSVP = require('rsvp');
 const Task = require('../models/task');
 const formatPackageList = require('../utilities/format-package-list');
 
+const logger = require('heimdalljs-logger')('ember-cli:tasks:bower-install');
+
 const Promise = RSVP.Promise;
 const resolve = RSVP.denodeify(require('resolve'));
+const writeJson = RSVP.denodeify(fs.writeJson);
 
 const cliPath = path.resolve(`${__dirname}/../..`);
 
 class BowerInstallTask extends Task {
 
+  resolveBower() {
+    logger.info('Resolving "bower" from %s ...', cliPath);
+    return resolve('bower', { basedir: cliPath });
+  }
+
+  importBower(path) {
+    logger.info('Importing "bower" from: %s', path);
+    this.bower = require(path);
+  }
+
   ensureBower() {
     if (this.bower) {
-      return Promise.resolve(this.bower);
+      return Promise.resolve();
     }
 
-    return resolve('bower', { basedir: cliPath })
-      .catch(() => this.installBower()
-        .then(() => resolve('bower', { basedir: cliPath })))
-      .then(bowerPath => require(bowerPath));
+    return this.resolveBower()
+      .catch(error => {
+        if (error.message.indexOf('Cannot find module \'bower\'') === -1) {
```

Comment:
tl;dr this is the best we can do for now... (see https://github.com/substack/node-resolve/commit/c622aefeb286e479d536601e30bb828e69f86ec3#commitcomment-20625328)